src/error.rs

```diff
@@ -1,11 +1,19 @@
+
 use defines::AfError;
 use std::error::Error;
+use std::sync::Mutex;
 
 pub type ErrorCallback = Fn(AfError);
 
 pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
```

Comment:
Hm, i think i found the root cause of this. `Fn(AfError)` needs to have Send and Sync traits implemented for it to be used either in the context of `Box` or `static mut`.

I am getting the error explained in the output of `rustc --explain E0117` if try to impl implement Send/Sync.